playbooks/roles/ocp-sso/tasks/main.yaml

```diff
@@ -0,0 +1,232 @@
+
+  - name: Verification of operator installation
+    block:
+    - name: Check if the cluster service version has succeeded
+      shell: oc get csv -n openshift-secondary-scheduler-operator --no-headers | awk '{ if (($1 ~ /^secondaryscheduleroperator/) && $NF=="Succeeded") print $1 }'| wc -l
+      register: sso_operators_csv
+      until: sso_operators_csv.stdout|int == 1
+      retries: 5
+      delay: 30
+
+    - name: Check if pods are running
+      shell: oc get pods -n openshift-secondary-scheduler-operator --no-headers | grep "scheduler-operator" | wc -l
+      register: sso_pods
+      until: sso_pods.stdout|int == 1
+      retries: 5
+      delay: 30
+
+- name: Deploy Secondary Scheduler 
+  block:
+  - name: Deploy the ConfigMap "secondary-scheduler-config" for configuring the scheduler-plugin profile for test
+    kubernetes.core.k8s:
+      state: present
+      definition:
+        apiVersion: v1
+        kind: ConfigMap
+        metadata:
+          name: "secondary-scheduler-config"
+          namespace: "openshift-secondary-scheduler-operator"
+        data:
+          "config.yaml": |
+            apiVersion: kubescheduler.config.k8s.io/v1
+            kind: KubeSchedulerConfiguration
+            leaderElection:
+              leaderElect: false
+            profiles:
+              - schedulerName: secondary-scheduler
+                plugins:
+                  score:
+                    disabled:
+                      - name: NodeResourcesBalancedAllocation
+                      - name: NodeResourcesLeastAllocated
+
+  - name: Create Secondary scheduler 
+    kubernetes.core.k8s:
+      state: present
+      definition:
+        apiVersion: operator.openshift.io/v1
+        kind: SecondaryScheduler
+        metadata:
+          name: cluster
+          namespace: openshift-secondary-scheduler-operator
+        spec:
+          logLevel: Normal
+          operatorLogLevel: Normal
+          managementState: Managed
+          schedulerConfig: secondary-scheduler-config
+          schedulerImage: "{{ sso_schedulerImage }}"
+
+  - name: Verification secondary scheduler pods
+    block:
+    - name: Check the secondary-scheduler pods running
+      shell: oc get pods -n openshift-secondary-scheduler-operator --no-headers | grep "Running" | wc -l
+      register: sso_pods
+      until: sso_pods.stdout|int == 2
+      retries: 5
+      delay: 10
+
+    - name: Check if pods are running
+      shell: oc get deploy -n openshift-secondary-scheduler-operator --no-headers | wc -l
+      register: sso_pods
+      until: sso_pods.stdout|int == 2
+      retries: 5
+      delay: 10
+
+- name: Run e2e test cases
+  block:
+  - name: Install dependencies for running e2e tests
+    dnf:
+      name: make
+      state: present
+
+  - name: Include role for installation of Go lang
+    include_role:
+      name: golang-installation
+    vars:
+      go_tarball: "{{ sso_go_tarball }}"
+      golang_path: "/usr/local"
+
```

Comment:
you can add a precheck task to check the the sso_github_username and sso_github_token variable are not None.

Reply:
Added a task to ensure the above variables are not empty if running the E2E tests.